streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -546,14 +547,14 @@ public void closeAndRecycleState() {
      *                 otherwise, just close open resources
      * @throws TaskMigratedException if the task producer got fenced (EOS)
      */
-    private Map<TopicPartition, Long> prepareClose(final boolean clean) {
-        final Map<TopicPartition, Long> checkpoint;
+    private void prepareClose(final boolean clean) {
+        // Reset any previously scheduled checkpoint.
```

Comment:
I think we could just reset any checkpoint at the beginning, so that we only do checkpointing if this call thinks so. @mjsax

Reply:
That should work.